services/audit-service/src/main/java/datawave/microservice/audit/AuditController.java

```diff
@@ -171,22 +184,33 @@ public String audit(@RequestParam MultiValueMap<String,String> parameters) {
             }
             
             if (log.isDebugEnabled())
-                log.debug("[" + restAuditParams.getAuditId() + "] Audit attempt " + attempts + " of " + retry.getMaxAttempts());
+                log.debug("[{}] Audit attempt {} of {}", auditParameters.getAuditId(), attempts, retry.getMaxAttempts());
             
-            success = sendMessage(restAuditParams);
+            success = sendMessage(auditParameters);
             currentTime = System.currentTimeMillis();
         } while (!success && (currentTime - auditStartTime) < retry.getFailTimeoutMillis() && attempts < retry.getMaxAttempts());
         
-        if (!success) {
-            log.warn("[" + restAuditParams.getAuditId() + "] Audit failed. { attempts = " + attempts + ", elapsedMillis = " + (currentTime - auditStartTime)
-                            + "}");
-            
-            throw new RuntimeException("Unable to process audit message with id [" + restAuditParams.getAuditId() + "]");
-        } else {
-            log.info("[" + restAuditParams.getAuditId() + "] Audit successful. { attempts = " + attempts + ", elapsedMillis = " + (currentTime - auditStartTime)
-                            + "}");
-            
-            return restAuditParams.getAuditId();
+        // last ditch effort to write the audit message to fileSystem for subsequent processing
+        if (!success && fileAuditor != null) {
+            success = true;
+            try {
+                if (log.isDebugEnabled())
```

Comment:
Unless the method call `auditParameters.getAuditId()` is expensive, there's no need to guard this statement with `if (log.isDebugEnabled())`

Reply:
updated.